sklearn/compose/_column_transformer.py

```diff
@@ -85,12 +85,11 @@ class ColumnTransformer(_BaseComposition, TransformerMixin):
         estimator must support `fit` and `transform`.
 
     sparse_threshold : float, default = 0.3
-        If the transformed output consists of a mix of sparse and dense data,
-        it will be stacked as a sparse matrix if the density is lower than this
-        value. Use ``sparse_threshold=0`` to always return dense.
-        When the transformed output consists of all sparse or all dense data,
-        the stacked result will be sparse or dense, respectively, and this
-        keyword will be ignored.
+        If the transformed output contains sparse matrices, it will be stacked
+        as a sparse matrix if the density is lower than this value. Use
+        ``sparse_threshold=0`` to always return dense.  When the transformed
+        output consists of all dense data, the stacked result will be dense,
+        and this keyword will be ignored.
```

Comment:
Should we also lift this special case? (the case of all dense data) 
In principle for the default of 0.3, this will always be the case anyway (you would need to set the threshold to something bigger than 1 to always have sparse)

Reply:
So you're saying we should consider a way to force dense output into sparse? I think we can leave that feature till later, as currently threshold > 1 is not valid.

Reply:
I feel like this case would be much rarer? When would you want that?

Reply:
I don't think I would ever want that, I was just wondering for simplicity's sake (to not have special cases). But maybe it is so logical to keep all dense as dense, that actually strictly following the sparse_threshold would actually be less "normal". 
So you can ignore my comment :)